lumicks/pylake/kymo.py

```diff
@@ -237,6 +246,26 @@ def plot_with_time_histogram(self, color_channel, pixels_per_bin=1, hist_ratio=0
         ax_hist.set_ylabel("counts")
         ax_hist.set_title(self.name)
 
+    def downsampled_by(self, downsampling_factor, reduce=np.sum):
+        result = Kymo(self.name, self.file, self.start, self.stop, self._json)
```

Comment:
I was originally thinking that this should be `result = copy(self)` which is a lot easier to read and understand. However, I wasn't sure what happens with the `._cache` attribute in that case. We may need to implement a custom `__copy__` to ensure correct behavior. I'd say it's not a priority for this PR but definitely something that should be done to ensure good, predictable behavior of the confocal classes. Both for us and any users that may be using these as part of other structures that might be copied.

Reply:
Agreed. I've made a ticket for it and bumped it to the top of the backlog.